### PR TITLE
[ARCTIC-1346][Core]: Clear kerberos ticket to access zk

### LINKED ISSUE #1346: [Bug]: When AMS is deployed HA, use Kerberos to access non-Kerberos zookeeper will throw an exception
### What happened?

When AMS is deployed HA, use Kerberos to access non-Kerberos zookeeper will throw an exception
image

### Affects Versions

0.3.2

### What engines are you seeing the problem on?

AMS

### How to reproduce

step1：AMS deployed HA
step2:  Use spark client to execute some SQL
step3:  Switch the ZK node
step4：Execute SQL again

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?

resolve #1346 
When AMS is deployed HA, use Kerberos to access non-Kerberos zookeeper will  throw an exception

image

## Brief change log

change in ArcticThriftUrl.parse()

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/ams-api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java [hunk 1]
```diff
@@ -74,17 +80,43 @@ public static ArcticThriftUrl parse(String url) {
...
-        } catch (Exception e) {
-          throw new RuntimeException("get master server info from zookeeper error", e);
+        int maxRetries = 3;
+        int retryCount = 0;
+        boolean success = false;
+
+        while (!success && retryCount < maxRetries) {
+          try {
+            ZookeeperService zkService = ZookeeperService.getInstance(zkServerAddress);
+            AmsServerInfo serverInfo = null;
```
Comment (Aireed) [inline ams/ams-api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java:90]: i think we can merge the two sentences to one!
> AmsServerInfo serverInfo = null;
            serverInfo = JSONObject.parseObject(
                zkService.getData(AmsHAProperties.getMasterPath(cluster)),
                AmsServerInfo.class);

**Review by Aireed — COMMENTED** [review]
I left  one comment , PTAL

**Review by zhoujinsong — COMMENTED** [review]
Thanks for your contribution!
After reviewing the code, I found areas that can be improved：

- The approach of obtaining server information from ZK, then concatenating the URL and further parsing it, seems unnecessary.
- The current parsing process code seems quite lengthy. It might be possible to split it into multiple methods to implement, such as using different methods for Thrift and Zookeeper protocols.
- Constants such as retry count should be declared as `private static final` at the beginning of the class.
- It is recommended to add necessary comments to the code explaining why authentication information is cleared when it appears, and any future improvement plans should be noted with TODO comments.

**Review by baiyangtx — APPROVED** [review]
LGTM

File: ams/ams-api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java [hunk 2]
```diff
@@ -113,13 +89,81 @@ public static ArcticThriftUrl parse(String url) {
...
+            }
+          }
+          return new ArcticThriftUrl("thrift",
+              serverInfo.getHost(), serverInfo.getThriftBindPort(), catalog.toLowerCase(), socketTimeout, url);
+        } catch (KeeperException.AuthFailedException authFailedException) {
+          // If kerberos authentication is not enabled on the zk,
+          // an error occurs when the thread carrying kerberos authentication information accesses the zk.
+          // Therefore, clear the authentication information and try again
+          retryCount++;
+          logger.error(String.format("Caught exception, retrying... (retry count: %s)", retryCount));
```
Comment (zhoujinsong) [inline ams/ams-api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java:144]: I'd recommend printing the exception stack here with `logger.error(msg,error)`

**Pull request merged**